### PR TITLE
fix for ES plugins install

### PR DESCRIPTION
**Pull request by dekhtyarev**
When using the ES plugins and restart(retry) the deployment, provided: **es_plugins_reinstall: false** we received an error when installing plugins, as their installation is not restricted (if defined).

I corrected this behavior, adding to check for **es_plugins_reinstall**.
But in this case we did not work the installation of plugins due to a bug in the variable **elasticsearch_install**.

Variable **elasticsearch_install** installation was recorded twice (feature ansible).
When you install from the repositories - we had the wrong variable **elasticsearch_install.changed**.

**elasticsearch_install** was divided into two different variables to correct this error.

### CODE REVIEW

**Comment by dekhtyarev** [comment]
Build failed on:
> TASK [elasticsearch : Copy templates to elasticsearch]

I did not change variables for this task.
Problems with ansible version on jenkins or am I wrong? 

@dliappis can you help me in this matter?

File: tasks/elasticsearch-Debian.yml [hunk 1]
```diff
@@ -19,4 +19,4 @@
 - name: Debian - Ensure elasticsearch is installed from downloaded package
   apt: deb=/tmp/elasticsearch-{{ es_version }}.deb
   when: not es_use_repository
-  register: elasticsearch_install
\ No newline at end of file
+  register: elasticsearch_install_from_package
```
Comment (antonbormotov) [inline tasks/elasticsearch-Debian.yml]: Please, add new line at the end

File: tasks/elasticsearch-RedHat.yml [hunk 2]
```diff
@@ -14,9 +14,9 @@
...
   when: es_use_repository
-  register: elasticsearch_install
+  register: elasticsearch_install_from_repo
 
 - name: RedHat - Install Elasticsearch from url
   yum: name={% if es_custom_package_url is defined %}{{ es_custom_package_url }}{% else %}{{ es_package_url }}-{{ es_version }}.noarch.rpm{% endif %} state=present
   when: not es_use_repository
-  register: elasticsearch_install
\ No newline at end of file
+  register: elasticsearch_install_from_package
```
Comment (antonbormotov) [inline tasks/elasticsearch-RedHat.yml]: +same

File: tasks/elasticsearch-Debian.yml (hunk 1 above)
Comment (dekhtyarev) [inline tasks/elasticsearch-Debian.yml]: added.

File: tasks/elasticsearch-RedHat.yml (hunk 2 above)
Comment (dekhtyarev) [inline tasks/elasticsearch-RedHat.yml]: added.

**Comment by bbaassssiiee** [comment]
This makes the role idempotent, great!

**Comment by gingerwizard** [comment]
@dekhtyarev LGTM.
I will fix the test fail issue now - i'm able to reproduce locally.  I will then merge. Apologies for the delayed response.

**Comment by gingerwizard** [comment]
@dekhtyarev merged in the changes through https://github.com/elastic/ansible-elasticsearch/pull/121 thanks for contribution!

**Pull request closed without merging**